### ISSUE TITLE
Hooks fail silently if wrong permission

### ISSUE THREAD

**Issue by TheLovinator1** [go-gitea/gitea#20771]
### Description

Hi,

**Context**:
I moved my Gitea files from Linux -> Windows -> Linux and lost my permissions.

**Problem**:
If the files in /git/repositories/username/gitrepo.git/hooks have the wrong file permissions (644 instead of 755) the hooks will fail without any errors. This breaks webhooks and no changes will be seen on the frontpage.

**What I expected**:
Gitea should change the permissions to the correct one or you should at least get a error message in the logs or failed delivery under `Recent Deliveries`.

### Gitea Version

1.18.0+dev-265-gc81b26b0e

### Git Version

2.36.2

### How are you running Gitea?

Running docker with the `gitea/gitea:dev` image

### Database

PostgreSQL

**Comment by lunny** [comment]
You could run `gitea doctor` to check and fix them, but it should not be automatically fixed.

**Comment by TheLovinator1** [comment]
How do I run gitea doctor on gitea/gitea:dev?

**gitea/gitea**:
`sudo docker exec -ti gitea-stable su git bash -c "cd /data/gitea/ && gitea doctor --fix"`

> [1] Check paths and basic configuration
>  - [I] Configuration File Path:    "/data/gitea/conf/app.ini"
>  - [I] Repository Root Path:       "/data/git/repositories"
>  - [I] Data Root Path:             "/data/gitea"
>  - [I] Custom File Root Path:      "/data/gitea"
>  - [I] Work directory:             "/app/gitea"
>  - [I] Log Root Path:              "/data/gitea/log"
> OK
> [2] Check Database Version
> OK
> [3] Check if user with wrong type exist
> OK
> [4] Check if OpenSSH authorized_keys file is up-to-date
> OK


**gitea/gitea:dev**:
`sudo docker exec -ti gitea-dev su git bash -c "cd /data/gitea/ && gitea doctor --fix"`
> 2022/08/12 21:10:02 main.go:118:main() [F] Failed to run app with [/usr/local/bin/gitea doctor --fix]: unable to init Git's HomeDir, incorrect initialization of the setting and git modules

**Comment by penlinux** [comment]
> panic: Failed to create sublogger (doctor): open doctor.log: permission denied

have a try:

```
docker exec -it --user git --workdir=/tmp gitea bash
```

```
gitea doctor --run hooks --fix
```

**Comment by TheLovinator1** [comment]
`docker exec -it --user git --workdir=/tmp gitea bash`
`gitea doctor --run hooks --fix`

> 2022/08/13 13:35:40 main.go:118:main() [F] Failed to run app with [/usr/local/bin/gitea doctor --run hooks --fix]: unable to init Git's HomeDir, incorrect initialization of the setting and git modules

This is what I get with a clean new version of gitea/gitea:dev with nothing on it, so it has broke since the last stable version.

**Comment by penlinux** [comment]
> 2022/08/13 13:35:40 main.go:118:main() [F] Failed to run app with [/usr/local/bin/gitea doctor --run hooks --fix]: unable to init Git's HomeDir, incorrect initialization of the setting and git modules

according to #20114 #20293  @wxiaoguang may know this issue

**Comment by wxiaoguang** [comment]
Actually, it's related to this one
* https://github.com/go-gitea/gitea/pull/20376

The `git.InitFull` expects that the the `settings.LoadConfXxx` has been called before. But in `doctor` sub-command, it hasn't.

I will take a look at about the config system + git module dependency.

**Comment by wxiaoguang** [comment]
The backport for 1.17: #20783
The fix for 1.18: #20782

**Comment by TheLovinator1** [comment]
Thanks, tried `gitea doctor --run hooks --fix` and the hooks are working again :)
